src/asmjit/core/zonestack.h

```diff
@@ -62,7 +62,7 @@ class ZoneStackBase {
     ASMJIT_INLINE_NODEBUG void setEnd(T* end) noexcept { _end = (void*)end; }
 
     template<typename T>
-    ASMJIT_INLINE_NODEBUG T* data() const noexcept { return (T*)((uint8_t*)(this) + sizeof(Block)); }
+    ASMJIT_INLINE_NODEBUG T* data() const noexcept { return (T*)((uint8_t*)const_cast<Block *>(this) + sizeof(Block)); }
```

Comment:
This feels wrong, I think it should be rather following to be const correct.
```suggestion
    ASMJIT_INLINE_NODEBUG T* data() const noexcept { return (T*)((uint8_t*)(this) + sizeof(Block)); }
    
    ASMJIT_INLINE_NODEBUG const T* data() const noexcept { return (const T*)((uint8_t*)(this) + sizeof(Block)); }
```

Reply:
I didn't set out to change the API, but sure, if that is preferred.

Reply:
In my opinion it is that or it should be left alone, having const_cast on a method that specifies const is not exactly good practice, this is however just my opinion I have no say in this matter and is only my suggestion.